linkerd/tracing/src/lib.rs

```diff
@@ -81,16 +81,18 @@ impl Settings {
         Some(Self {
             filter,
             format: std::env::var(ENV_LOG_FORMAT).ok(),
-            access_log: std::env::var(ENV_ACCESS_LOG).is_ok(),
+            access_log: std::env::var(ENV_ACCESS_LOG)
+                .ok()
+                .and_then(|env| env.parse().ok()),
```

Comment:
should we log/report parse failures?

Reply:
I guess this is currently infallible. can we use `Infallible` error type for the parser?

Reply:
this never fails, as any value is currently parsed as the default (apache common) format:
https://github.com/linkerd/linkerd2-proxy/blob/bc19360954c3cc42c3376b64b21c1ddb9e297f95/linkerd/tracing/src/access_log.rs#L168-L173

if we'd prefer the user to always have to explicitly specify "apache" (rather than accepting e.g. `LINKERD2_PROXY_ACCESS_LOG=1` or similar), i can change it to actually return a parse error that we can print, yeah.

Reply:
Actually, yeah; let's do that. this is going to have to be wired through from an annotation, so having a few documented, support values seems fine.

Reply:
okay, will do!

Reply:
changed it to only accept 'json' and 'apache' in 921cc58